maint-lib/makelib.mk

```diff
@@ -19,11 +19,11 @@ $(shell bash -c 'set -eu ; \
 	set_var CEPH_POINT_RELEASE "$$(bash maint-lib/ceph_version.sh "$$ceph_version_spec" CEPH_POINT_RELEASE)" ; \
 	set_var BASEOS_REPO        "$(word 3, $(subst $(comma), , $(1)))" ; \
 	set_var BASEOS_TAG         "$(word 4, $(subst $(comma), , $(1)))" ; \
-	set_var BASEOS_REGISTRY    "_" ; \
+	set_var BASEOS_REGISTRY    "$(BASEOS_REGISTRY)" ; \
 	set_var IMAGES_TO_BUILD    "$(IMAGES_TO_BUILD)" ; \
 
 	set_var STAGING_DIR       "staging/$$CEPH_VERSION$$CEPH_POINT_RELEASE-$$BASEOS_REPO-$$BASEOS_TAG-$$HOST_ARCH" ; \
-	base_img="$$BASEOS_REGISTRY/$$BASEOS_REPO:$$BASEOS_TAG" ; \
+	base_img="$$BASEOS_REPO:$$BASEOS_TAG" ; \
 	if [ -n "$(BASE_IMAGE)" ] ; then base_img="$(BASE_IMAGE)" ; fi ; \
```

Comment:
This will still overwrite `BASE_IMAGE` with the user-specified value without adding the registry to it.

I think that the whole code flow, starting from what I see as line 26 to line 28 just below is:

```sh
base_img="$$BASEOS_REPO:$$BASEOS_TAG" ; \  # Set to the default computed value for <repo>:<tag>
if [ -n "$(BASE_IMAGE)" ] ; then base_img="$(BASE_IMAGE)" ; fi ; \  # override with user-specified <repo>:<ta> if BASE_IMAGE was specified by user
base_img="$$BASEOS_REGISTRY/$$base_img" ; \  # attach the registry to the front of <repo>:<tag>
set_var BASE_IMAGE_FULL       "$${base_img#_/}"  # Strip the registry (and the '/') if the registry is default '_', and then set the value to the env var BASE_IMAGE_FULL (which used to be BASE_IMAGE)
```